testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -1050,12 +1050,12 @@ impl BlockMinerThread {
         let (mut block, consumed, size, tx_events) = NakamotoBlockBuilder::build_nakamoto_block(
             &chain_state,
             &burn_db
-                .index_handle_at_ch(&self.burn_block.consensus_hash)
```

Comment:
Question for reviewers:  I'm pretty sure this is buggy -- the sortition handle we want to give to `build_nakamoto_block` should represent the sortition history up through the burn election block, not the burn chain tip, right?

I don't think this is the complete fix, however.  If there's a tenure-extend, then we'd want to present a sortition history as of the last tenure-extend.

Reply:
Like, the view we want to give to the Clarity VM needs to be the sortition history up to the last `TenureChange` -- be it a `BlockFound` or a `TenureExtend`.

Reply:
Hmm I'm not sure. Is this change necessary? I'm not sure that `self.burn_block` _is_ the tip (ie in the case of a tenure extend), I'll have to look into it more in the morning.

Reply:
The burn_block is the block that triggered the miner thread. So in the case of a tenure extend, it will be a newer burn block then the election burn block of the original tenure.

Reply:
Yup, saw that and reverted. Thanks folks! :raised_hands: